src/scripts/main.js

```diff
@@ -1,3 +1,380 @@
 'use strict';
 
 // write code here
+function sortTableByColumn(columnIndex, sortParam, table) {
+  const tableCopy = [...table];
+
+  tableCopy.sort((a, b) => {
+    const cellA = a.cells[columnIndex].textContent;
+    const cellB = b.cells[columnIndex].textContent;
+
+    if (sortParam === 'Salary') {
+      const salaryA = +cellA.replace(/\D/g, '');
+      const salaryB = +cellB.replace(/\D/g, '');
+
+      return salaryA - salaryB;
+    }
+
+    return cellA.localeCompare(cellB);
+  });
+
+  if (sortField === sortParam && !descending) {
```

Comment:
The variable `sortField` is used here but is not defined within the function or passed as a parameter. Ensure that `sortField` is properly defined and accessible in this context.